src/vivarium/framework/results/context.py

```diff
@@ -0,0 +1,106 @@
+from collections import defaultdict
+from typing import Callable, Dict, List, Tuple
+
+import pandas as pd
+
+from vivarium.framework.results.exceptions import ResultsConfigurationError
+
+
+class ResultsContext:
+
+    def __init__(self):
+        self._default_stratifications = []  # type: List[str]
+        self._stratifications = []        # type: List[Stratification]
+        # keys are event names
+        # values are dicts with key (filter, grouper) value (measure, aggregator, additional_keys)
+        self._observations = defaultdict(lambda: defaultdict(list))
+
+    def set_default_stratifications(self, default_grouping_columns: List[str]):
+        if self._default_stratifications:
+            raise ResultsConfigurationError('Multiple calls are being made to set default grouping columns '
```

Comment:
What is the purpose of self._default_stratifications?

Reply:
The intent is for there to be support for default stratifications for observers, as there is now. Example in CVD: https://github.com/ihmeuw/vivarium_nih_us_cvd/blob/4e393f5687e3949ad278a5bd7b30d0efabf51ca6/src/vivarium_nih_us_cvd/model_specifications/nih_us_cvd.yaml#L64-L68